Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+                return $false
+            }   
+        }
+        catch {
+            return $false
+        }
+    }
+}
+
+Function Get-ServiceInstallPath {
+    <#
+    .SYNOPSIS
+    Get the installation path of a service
+    .DESCRIPTION
+    The function receive the service name and return the path or returns NULL if not found
+    .EXAMPLE
+    (Get-ServiceInstallPath $<ServiceName>) -ne $NULL
+    .PARAMETER ServiceName
+    The service name to query. Just one.
+    #>
+    param ($ServiceName)
+    Begin {
+        
+    }
+    Process {
+        $retInstallPath = $null
+        try {
+            Set-Variable -Name m_ServiceList -Value $(Get-ChildItem "HKLM:\System\CurrentControlSet\Services" | ForEach-Object { Get-ItemProperty $_.pspath }) -Scope Script
+            $regPath = $m_ServiceList | Where-Object { $_.PSChildName -eq $ServiceName }
+            If ($Null -ne $regPath) {
+                $retInstallPath = $regPath.ImagePath.Substring($regPath.ImagePath.IndexOf('"'), $regPath.ImagePath.LastIndexOf('"') + 1)
+            }
+        }
+        catch {
+            Throw $(New-Object System.Exception ("Cannot get Service Install path for $ServiceName", $_.Exception))
+        }
+        
+        return $retInstallPath
+    }
+    End {
+        
+    }
+}
+
+Function New-ConnectionToRestAPI {
+    <#
+    .SYNOPSIS
+    Get the installation path of a service
+    .DESCRIPTION
+    The function receive the service name and return the path or returns NULL if not found
+    .EXAMPLE
+    (Get-ServiceInstallPath $<ServiceName>) -ne $NULL
+    .PARAMETER pvwaAddress
+    The PVWA server address (e.g. https://subdomain.privilegecloud.cyberark.cloud)
+    .PARAMETER tinaCreds
+    Tenant administrator/installer user credentials
+    #>
+    # Get PVWA and login informatioN
+    param (
+        [Parameter(Mandatory = $true)]
+        $pvwaAddress,
+        [Parameter(Mandatory = $true)]
+        [PSCredential]$tinaCreds        
+    )
+    $url = $pvwaAddress + "/PasswordVault/API/auth/Cyberark/Logon"
+    $BSTR = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($tinaCreds.Password)
+    
+    $headerPass = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto($BSTR)
+    $body = @{
+        username = $tinaCreds.UserName
+        password = $headerPass
+    }
+    $json = $body | ConvertTo-Json
+    Try {
+        $pvwaToken = Invoke-RestMethod -Method 'Post' -Uri $url -Body $json -ContentType 'application/json'
+    }
+    Catch {
+        Write-Host "Failed to retrieve token. Response received:"
+        Write-Host $_.Exception.Message
+        exit 1
+    }
+    if ($pvwaToken -match "[0-9a-zA-Z]{200,256}") {
+        return $pvwaToken
+    }
+    else {
+        Write-Host "Failed to retrieve token. Response received:"
+        Write-Host $_.Exception.Message
+        exit 1
+    }
+}
+
+Function Test-PvwaToken {
+    <#
+    .SYNOPSIS
+    Test a PVWA token to ensure it is valid
+    .DESCRIPTION
```

Comment:
Function to test PVWA token by attempting to retrieve a list of accounts (this will be successful even if there are no accounts available)